packages/qdrant-loader/src/qdrant_loader/connectors/confluence/connector.py

```
@@ -792,162 +804,249 @@ def _clean_html(self, html: str) -> str:
-                        logger.error(f"Failed to parse next URL: {e!s}")
+                    parsed_url = urlparse(next_url)
+                    query_params = parse_qs(parsed_url.query)
+                    cursor = query_params.get("cursor", [None])[0]
+                    if not cursor:
+                        logger.debug(
+                            "No cursor found in next URL, ending pagination"
+                        )
                         break
-
+                    logger.debug(f"Found next cursor: {cursor}")
                 except Exception as e:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)